### PR TITLE
Add weekly workflow to keep Tested up to current

### PR DESCRIPTION
**Pull request by obenland**
Adds a weekly GitHub Actions workflow that:

- Runs every Monday at 09:00 UTC (and via manual dispatch).
- Fetches the latest stable WordPress version from api.wordpress.org.
- Updates `Tested up to:` in readme.txt if it's behind.
- Opens a PR with the bump and auto-merges it (squash).

Goal: keep `Tested up to` corresponding to the latest stable WordPress version, fully automatically.

### CODE REVIEW

File: .github/workflows/update-tested-up-to.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    update:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v6
+
+            - name: Get latest WordPress version
+              id: wp
+              run: |
+                  latest=$(curl -fsSL https://api.wordpress.org/core/version-check/1.7/ | jq -r '.offers[0].version')
+                  short=$(echo "$latest" | awk -F. '{print $1"."$2}')
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml:33]: Fixed — both bash steps now use `set -euo pipefail`, and the fetched WordPress version is validated against `^[0-9]+\.[0-9]+(\.[0-9]+)?$` before it's used. The job fails if `curl`/`jq` errors or the API returns anything unexpected, so a malformed value can no longer reach `readme.txt`.

File: .github/workflows/update-tested-up-to.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                  echo "Latest WordPress: $latest (short: $short)"
+
+            - name: Update readme.txt
+              id: update
+              run: |
+                  current=$(grep -i '^Tested up to:' readme.txt | head -1 | sed -E 's/^[Tt]ested up to:[[:space:]]+//' | tr -d '\r')
+                  if [ "$current" = "${{ steps.wp.outputs.version }}" ]; then
+                      echo "Already on latest ($current); nothing to do."
+                      echo "updated=false" >> "$GITHUB_OUTPUT"
+                      exit 0
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml]: Fixed — the step now fails explicitly if `current` is empty (missing/malformed `Tested up to:` header) instead of proceeding with an empty `previous`.

File: .github/workflows/update-tested-up-to.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                  fi
+                  sed -i -E "s/^([Tt]ested up to:[[:space:]]+).*/\1${{ steps.wp.outputs.version }}/" readme.txt
+                  {
+                      echo "updated=true"
+                      echo "previous=$current"
+                  } >> "$GITHUB_OUTPUT"
+                  echo "Bumped Tested up to: $current -> ${{ steps.wp.outputs.version }}"
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml]: Fixed — after the `sed` substitution, the step re-reads the `Tested up to:` line and fails if the resulting value doesn't equal `target`. `updated=true` is only emitted when the file was genuinely modified.

File: .github/workflows/update-tested-up-to.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+jobs:
+    update:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v6
+
+            - name: Get latest WordPress version
+              id: wp
+              run: |
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml:26]: Fixed — both bash steps now use `set -euo pipefail`, and the fetched WordPress version is validated against `^[0-9]+\.[0-9]+(\.[0-9]+)?$` before it's used. The job fails if `curl`/`jq` errors or the API returns anything unexpected, so a malformed value can no longer reach `readme.txt`.

File: .github/workflows/update-tested-up-to.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+concurrency:
+    group: update-tested-up-to
+    cancel-in-progress: false
+
+jobs:
+    update:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v6
+
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml:22]: Not applied here — none of the existing workflows in this repo (or its sibling plugin repos) pin actions to commit SHAs; they all use floating major tags. Pinning just this one workflow would be inconsistent. I've flagged a separate cross-repo hardening pass to do this consistently along with Renovate/Dependabot wiring; doing it piecemeal would create maintenance churn without the version-update automation to balance it.

File: .github/workflows/update-tested-up-to.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            - name: Get latest WordPress version
+              id: wp
+              shell: bash
+              run: |
+                  set -euo pipefail
+                  latest=$(curl -fsSL https://api.wordpress.org/core/version-check/1.7/ | jq -r '.offers[0].version')
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml]: Fixed — added `--connect-timeout 30 --max-time 60 --retry 3 --retry-delay 2` to the `curl` call so a stalled api.wordpress.org fails fast instead of hanging until the runner-level timeout.

File: .github/workflows/update-tested-up-to.yml [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                  {
+                    echo "updated=true"
+                    echo "previous=$current"
+                  } >> "$GITHUB_OUTPUT"
+                  echo "Bumped Tested up to: $current -> $target"
+
+            - name: Create Pull Request
+              if: steps.update.outputs.updated == 'true'
+              id: pr
+              uses: peter-evans/create-pull-request@v7
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml]: Not applied here — none of the existing workflows in this repo (or its sibling plugin repos) pin actions to commit SHAs; they all use floating major tags. Pinning just this one workflow would be inconsistent. I've flagged a separate cross-repo hardening pass to do this consistently along with Renovate/Dependabot wiring; doing it piecemeal would create maintenance churn without the version-update automation to balance it.

File: .github/workflows/update-tested-up-to.yml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                  {
+                      echo "updated=true"
+                      echo "previous=$current"
+                  } >> "$GITHUB_OUTPUT"
+                  echo "Bumped Tested up to: $current -> $target"
+
+            - name: Create Pull Request
+              if: steps.update.outputs.updated == 'true'
+              id: pr
+              uses: peter-evans/create-pull-request@v7
```
[the remark this answers is not in the document]
Comment (obenland) [reply .github/workflows/update-tested-up-to.yml]: Not applied — the `GITHUB_TOKEN` chain-trigger limitation no longer matters for this automation. The bump now reaches WordPress.org via the inline `10up/action-wordpress-plugin-asset-update@stable` step at the end of this workflow, so we do not depend on `push-asset-readme-update.yml` firing on the post-merge push. End-to-end fully automatic, no PAT or App-token setup required.
